docs/RFCS/20201020_sql_query_denylist.md

```diff
@@ -0,0 +1,183 @@
+- Feature Name: SQL Query Denylist
+- Status: draft
+- Start Date: 2020-10-20
+- Authors: angelapwen, otan
+- RFC PR: (PR # after acceptance of initial draft)
+- Cockroach Issue: [51643](https://github.com/cockroachdb/cockroach/issues/51643)
+
+# Summary
+
+We propose a mechanism built into CockroachDB to prevent operator-defined SQL queries from being executed. This will immediately address the situations where clusters enter unresponsive states due to a query, or a query pattern, that has been identified. To bring the clusters back up as soon as possible and prevent them from going down again due to the same query, we need a query denylist on the database. This RFC details a file-managed SQL query denylist implementation that will allow database administrators to trigger query denial by uploading a YAML file comprising regular expression patterns to a customizable file location.  
```

Comment:
DBAs would need file write permission to do this. What user/level of access do CC SREs have to CC clusters?

Reply:
I don't know the answer to this — perhaps an SRE reviewing here could answer? @joshimhoff or @JuanLeon1?